pkg/agent/definitions.go

```diff
@@ -90,11 +90,55 @@ type KubeConfig struct {
 	InsightRefreshDuration string
 }
 
+type KubeAgentMonitorConfig struct {
+	AgentNamespace      string
+	Daemonset           string
+	Deployment          string
+	DaemonsetConfigMap  string
+	DeploymentConfigMap string
+}
+
+// WithDaemonset sets the daemonset name for the agent
+func WithDaemonset(v string) KubeAgentMonitorOptions {
```

Comment:
Name these `WithKubeAgentMonitorDaemonset`, `WithKubeAgentMonitorDeployment` etc. This makes it clear that these should only be used Kube Agent Monitor.